src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3677,13 +3677,48 @@ public override Type MakeArrayType(int rank)
         #region Invoke Member
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern bool CanValueSpecialCast(RuntimeType valueType, RuntimeType targetType);
+        private static extern bool CanPrimitiveWiden(CorElementType valueType, CorElementType targetType);
+
+        private static bool CanValueSpecialCast(RuntimeType valueType, RuntimeType targetType)
+        {
+            Debug.Assert(targetType.IsPointer || targetType.IsEnum || targetType.IsPrimitive || targetType.IsFunctionPointer);
+
+            CorElementType targetCorElement = RuntimeTypeHandle.GetVerifierCorElementType(targetType);
+            if (targetCorElement is CorElementType.ELEMENT_TYPE_PTR or CorElementType.ELEMENT_TYPE_FNPTR)
```

Comment:
Can this use property checks on `Type` instead?

Reply:
I had converted this exact method with no new FCall.

Reply:
See comments by @jkotas and @am11 above.